ov/connection_templates.go

```diff
@@ -0,0 +1,147 @@
+	"github.com/docker/machine/libmachine/log"
+)
+
+type BandwidthType struct {
+	MaximumBandwidth int `json:"maximumBandwidth,omitempty"`
+	TypicalBandwidth int `json:"typicalBandwidth,omitempty"`
+}
+
+type ConnectionTemplate struct {
+	Bandwidth   BandwidthType `json:"bandwidth,omitempty"`
+	Category    string        `json:"category,omitempty"`
+	Created     string        `json:"created,omitempty"`
+	Description utils.Nstring `json:"description,omitempty"`
+	ETAG        string        `json:"eTag,omitempty"`
+	Modified    string        `json:"modified,omitempty"`
+	Name        string        `json:"name,omitempty"`
+	State       string        `json:"state,omitempty"`
+	Status      string        `json:"status,omitempty"`
+	Type        string        `json:"type,omitempty"`
+	URI         utils.Nstring `json:"uri,omitempty"`
+}
+
+type ConnectionList struct {
+	Category    string               `json:"category,omitempty"`
+	Count       int                  `json:"count,omitempty"` // "count": 1,
+	Created     string               `json:"created,omitempty"`
+	ETAG        string               `json:"eTag,omitempty"`
+	Members     []ConnectionTemplate `json:"members,omitempty"`
+	Modified    string               `json:"modified,omitempty"`
+	NextPageURI utils.Nstring        `json:"nextPageUri,omitempty"` // "nextPageUri": null,
+	PrevPageURI utils.Nstring        `json:"prevPageUri,omitempty"` // "prevPageUri": null,
+	Start       int                  `json:"start,omitempty"`       // "start": 0,
+	Total       int                  `json:"total,omitempty"`       // "total": 1
+	Type        string               `json:"type,omitempty"`
+	URI         utils.Nstring        `json:"uri,omitempty"`
+}
+
+func (c *OVClient) GetConnectionTemplateByName(name string) (ConnectionTemplate, error) {
+	conntemplate, err := c.GetConnectionTemplate(fmt.Sprintf("name matches '%s'", name), "name:asc", "", "")
+	if conntemplate.Total > 0 {
+		return conntemplate.Members[0], err
+	}
+
+	return ConnectionTemplate{}, err
+}
+
+func (c *OVClient) GetConnectionTemplate(filter string, sort string, start string, count string) (ConnectionList, error) {
+	var (
+		uri            = "/rest/connection-templates"
+		q              = make(map[string]interface{})
+		connectionlist ConnectionList
+	)
+	if len(filter) > 0 {
+		q["filter"] = filter
+	}
+
+	if sort != "" {
+		q["sort"] = sort
+	}
+
+	if start != "" {
+		q["start"] = start
+	}
+
+	if count != "" {
+		q["count"] = count
+	}
+
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+	// Setup query
+	if len(q) > 0 {
+		c.SetQueryString(q)
+	}
+	data, err := c.RestAPICall(rest.GET, uri, nil)
+	if err != nil {
+		return connectionlist, err
+	}
+
+	log.Debugf("GetConnectionTemplate %s", data)
+	if err := json.Unmarshal(data, &connectionlist); err != nil {
+		return connectionlist, err
+	}
+	return connectionlist, nil
+}
+
+func (c *OVClient) UpdateConnectionTemplate(conntemplate ConnectionTemplate) error {
```

Comment:
This function should provide a response body of type ConnectionTemplate and error